src/category_theory/yoneda.lean

```diff
@@ -127,6 +127,13 @@ If `coyoneda.map f` is an isomorphism, so was `f`.
 def is_iso {X Y : Cᵒᵖ} (f : X ⟶ Y) [is_iso (coyoneda.map f)] : is_iso f :=
 is_iso_of_fully_faithful coyoneda f
 
+-- No need to use Cᵒᵖ here, works with any category
+/-- A Type-valued presheaf `P` is isomorphic to the composition of `P` with the
+  coyoneda functor coming from `punit`. -/
+def iso_comp_punit (P : C ⥤ Type v₁) : P ≅ (P ⋙ coyoneda.obj (op punit.{v₁+1})) :=
```

Comment:
Could you add `simps` to this? Also I feel like this is more useful in the opposite direction.

Reply:
I can do that, but I don't know what `simps` means and I don't know how to find out (I'm sure I've been told before but I need to be told something 5 times nowadays before I learn it, and it's easier to look it up 5 times rather than ask 5 times). Where does one look up meanings of attributes? Is this something to do with the category theory version of x*(x^{-1}*y)=y?

Reply:
Not exactly (you might be thinking of `reassoc`). `simps` generates projection lemmas, eg the one called `iso_comp_punit_hom_app` which looks like `(iso_comp_punit P).hom.app X = ... := rfl` and tags them simp. Essentially it gives us definitional unfolding simp-lemmas which is useful when you need to prove things where this iso is in the goal.